docs/EN_US/ECLProgrammersGuide/PRG_Mods/PrGit_Source_Code.xml

```diff
@@ -0,0 +1,620 @@
+      user interaction, such as passphrases for ssh keys, web interaction with
+      GitHub authentication, and initial entry for cached access tokens. This
+      is problematic for the ECLCC Server which cannot support user
+      interaction, and since it is preferrable not to pass credentials around.
+      The solution therefore is to use a personal access token securely stored
+      as a secret. This token could then be associated with a special service
+      account, which would then securely initiate these transactions. The
+      secret then avoids the need to pass credentials and allows the keys to
+      be rotated. </para>
+    </sect3>
+
+    <sect3 id="PrgKubernetesSecrets">
+      <title>Kubernetes Secrets</title>
+
+      <para>This section describe secrets support in the Kubernetes (and Bare
+      Metal) versions of the HPCC Systems platform. </para>
+
+      <para>To add secrets support:</para>
+
+      <orderedlist>
+        <listitem>
+          <para>. Add the gitUsername property to the eclccserver component of
+          your customization yaml file:</para>
+
+          <programlisting>   eclccserver:
+   - name: myeclccserver
+     gitUsername: <emphasis role="bold">gituser</emphasis>
+</programlisting>
+
+          <para>Note: the eclccserver.gitUsername value should match your git
+          user name. </para>
+        </listitem>
+
+        <listitem>
+          <para>Add a secret to the customization yaml file, with a key that
+          matches the gitUsername</para>
+
+          <programlisting>secrets:
+     git: 
+       gituser: <emphasis role="bold">my-git-secret</emphasis>
+</programlisting>
+
+          <para></para>
+        </listitem>
+
+        <listitem>
+          <para>Add the secret to Kubernetes containing the personal access
+          token:</para>
+
+          <programlisting>apiVersion: v1
+kind: Secret
+metadata:
+  name: <emphasis role="bold">my-git-secret</emphasis>
+type: Opaque
+stringData:
+  password: ghp_eZLHeuoHxxxxxxxxxxxxxxxxxxxxol3986sS=
+</programlisting>
+
+          <para>Note password contains the personal access token.</para>
+        </listitem>
+
+        <listitem>
+          <para>Apply the secret to your Kubernetes using the kubectl command:
+          </para>
+
+          <programlisting>kubectl apply -f ~/dev/hpcc/helm/secrets/my-git-secret</programlisting>
+
+          <para>When a query is submitted to the ECLCC Server, any git
+          repositories are then accessed using this configured user name and
+          password. </para>
+        </listitem>
+
+        <listitem>
+          <para>Store the secret in a vault. You can also store the PAT
+          (personal access token) inside a vault.</para>
+        </listitem>
+      </orderedlist>
+
+      <para>and that is how we add the secrets to a cloud deployment. </para>
+    </sect3>
+
+    <sect3 id="CredentialsInBareMetal">
+      <title>Bare Metal Credentials</title>
```

Comment:
the term bare metal has been written in a few different ways. Bare-metal, bare metal, etc. Need to pick one and be consistent.